CMakeLists.txt

```diff
@@ -126,6 +126,7 @@ elseif(BASEARCH_X86_FOUND)
     option(WITH_SSE2 "Build with SSE2" ON)
     cmake_dependent_option(WITH_SSSE3 "Build with SSSE3" ON "WITH_SSE2" OFF)
     cmake_dependent_option(WITH_SSE42 "Build with SSE42" ON "WITH_SSSE3" OFF)
+    cmake_dependent_option(WITH_SSE41 "Build with SSE41" ON "WITH_SSSE3" OFF)
```

Comment:
```suggestion
    cmake_dependent_option(WITH_SSE41 "Build with SSE41" ON "WITH_SSSE3" OFF)
    cmake_dependent_option(WITH_SSE42 "Build with SSE42" ON "WITH_SSE41" OFF)
```

Reply:
So things do get a little strange here - namely there are CPUs with SSE4a that do not have SSSE3 (I think AMD fits that bill for a lot of those).  However, I think for the full 4.1 set this is correct in that 4.2 implies 4.1.

Reply:
I didn't understand you.
SSE4a is a separate instruction set from AMD and it has a dedicated CPUID bit.

This changes use SSE4.1 and SSE4.1 CPUID bit.

Reply:
These checks are added to prevent the user from building an impossible configuration.
For example, with SSE4.2 but without SSE2 (ran cmake with flags ``-DWITH_SSE2=OFF -DWITH_SSE42=ON``).